concent_api/core/utils.py

```diff
@@ -118,3 +123,34 @@ def extract_name_from_scene_file_path(absoulte_scene_file_path_in_docker: str) -
             ErrorCode.MESSAGE_INVALID
         )
     return relative_scene_file_path_in_archive
+
+
+def is_protocol_verison_compatible_with_verison_in_concent(protocol_version: str) -> bool:
+    """
+    Versions are considered compatible if they share the minor and major version number.
+    E.g 2.18.5 is compatible with 2.18.1 but not with 2.17.5 or 3.0.0.
+    """
+    return protocol_version.split('.')[0] == settings.GOLEM_MESSAGES_VERSION.split('.')[0] and \
+        protocol_version.split('.')[1] == settings.GOLEM_MESSAGES_VERSION.split('.')[1]
+
+
+def if_given_version_of_golem_messages_is_compatible_with_version_in_concent(
+    request: HttpRequest,
+    client_public_key: Optional[bytes] = None,
+) -> bool:
+    """
+    If header is missing version is not checked and Concent assumes that client uses compatible version.
+    """
+    if 'HTTP_CONCENT_GOLEM_MESSAGES_VERSION' not in request.META:
+        return True
+    else:
+        golem_message_version = request.META['HTTP_CONCENT_GOLEM_MESSAGES_VERSION']
+        if not is_protocol_verison_compatible_with_verison_in_concent(golem_message_version):
+            log(
```

Comment:
Utils should not log stuff. After removing this it is only needed to do:
```
return is_protocol_verison_compatible_with_verison_in_concent(golem_message_version)
```

Reply:
This is called in views and decorators. I think it should be logged in general and where to log.. I think it is good to hide this here instead of log in view function. Is it OK as is is now or change it and move logging above?

Reply:
But this is utility function and might be used elsewhere, and logging elsewhere might not be required or confusing. The view should log or not depending on the value returned from this function.

Reply:
done